src/main/java/bisq/price/util/cryptoya/CryptoYaMarketData.java

```diff
@@ -32,9 +36,30 @@ public class CryptoYaMarketData {
     private CryptoYaRate bybit;
     private CryptoYaRate binance;
 
-    public Double averageBluePrice() {
-        // TODO implement average of reported values
-        // consider removing old prices (refresh date comes in UTC)
-        return argenbtc.getTotalBid();
+    /**
+     *
+     * @return the avg ask price from all the exchanges that have updated ask prices (not older than 1 day)
+     *      if anything goes wrong defaults to the first available ask price, and if no data is avaialble return 0.0
+     */
+    public Double averageBlueRate() {
+        // filter more than 1 day old values with yesterday UTC timestamp
+        final Long yesterdayTimestamp = Instant.now().minus(1, ChronoUnit.DAYS).getEpochSecond();
+        final OptionalDouble average = streamUsefulAvailableMarkets(yesterdayTimestamp).mapToDouble(CryptoYaRate::getAsk).average();
```

Comment:
This can be reduced to:
```
return streamUsefulAvailableMarkets(yesterdayTimestamp)
                .mapToDouble(CryptoYaRate::getAsk)
                .average()
                .orElse(0);
```

If the stream is empty average is empty as well, so it leads to the same result as in your code.

Reply:
Good suggestion @HenrikJannsen , I'll add it in my next commit!